plugins/modules/net_tools/nmcli.py

```diff
@@ -91,10 +91,45 @@
     routes4:
         description:
             - The list of ipv4 routes.
-            - Use the format '192.0.3.0/24 192.0.2.1'
         type: list
-        elements: str
+        elements: dict
```

Comment:
This change is problematic, because it is a breaking change. Current uses of these parameters will fail once this is merged. That's why we cannot merge this (as-is).

What is possible:
1. Make this `elements: raw` and write code to make sure that these are either strings or dictionaries in the exact format you require.
2. Alternatively, add new options that are lists of dicts and make these mutually exclusive with the old options.

Both alternatives aren't great. 2. allows to properly document the format for the new options, while 1. does not allow to do this (`suboptions` only work for `type=dict` or `type=list`/`elements=dict`).

Reply:
I really like the second idea, but should we make new option mutually exclusive? Maybe i can make them work simultaneously.

Reply:
Not making it mutually exclusive usually makes the code more complicated, because then you somehow have to join the two lists, care about ordering, etc. If it's just one or the other, there's no ambiguity what can happen.